### PR TITLE
Corrected code for test case :CancelAllRunningBackupJobs

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
Backup were not getting deleted even after 20 mins of wait.
Have modified the code and added additional check for backup deletion

**Which issue(s) this PR fixes** (optional)
Closes #https://portworx.atlassian.net/browse/PA-744

**Special notes for your reviewer**:
Pass log : https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/890/consoleFull

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/898/consoleFull

### CODE REVIEW

File: tests/backup/backup_resiliency_test.go [hunk 1]
```diff
@@ -708,22 +708,43 @@ var _ = Describe("{CancelAllRunningBackupJobs}", func() {
...
+						if actual == requiredStatus {
+							return "", false, nil
+						}
+						if actual == expectedStatus {
+							return "", true, fmt.Errorf("backup status for [%s] expected was [%v] but got [%s]", backupName, requiredStatus, actual)
+						} else {
+							return "", false, fmt.Errorf("backup status for [%s] expected was [%v] but got [%s]", backupName, requiredStatus, actual)
+						}
+					}
+					_, err = task.DoRetryWithTimeout(backupProgressCheckFunc, maxWaitPeriodForBackupJobCancellation*time.Minute, backupJobCancellationRetryTime*time.Second)
```
Comment (ak-px) [inline tests/backup/backup_resiliency_test.go:745]: It's better to reduce the retry time from 30s to a lower value.

File: tests/backup/backup_resiliency_test.go [hunk 2]
```diff
@@ -750,12 +771,13 @@ var _ = Describe("{CancelAllRunningBackupJobs}", func() {
 					}
 					return "", false, nil
 				}
-				_, err = task.DoRetryWithTimeout(backupJobCancelStatus, maxWaitPeriodForBackupJobCancellation*time.Minute, backupJobCancellationRetryTime*time.Second)
+				_, err = task.DoRetryWithTimeout(backupJobCancelStatus, backupDeleteTimeout*time.Minute, backupDeleteRetryTime*time.Second)
 				if err != nil {
 					adminBackups, error1 := GetAllBackupsAdmin()
-					log.Infof("The list of backups still present after backup cancellation and wait of 10 minutes is %v,Error:%v", adminBackups, err)
 					log.FailOnError(error1, "Getting the list of backups after backup cancellation and wait of 10 minutes")
+					log.Infof("The list of backups still present after backup cancellation and wait of 10 minutes is %v,Error:%v", adminBackups, err)
```
Comment (ak-px) [inline tests/backup/backup_resiliency_test.go:778]: The log need to be edited, The wait minute is not static.

**Review by ak-px — APPROVED** [review]
LGTM, apart from minor log change.

File: tests/backup/backup_resiliency_test.go (hunk 2 above)
Comment (snigdha-px) [reply tests/backup/backup_resiliency_test.go]: corrected

**Review by kshithijiyer-px — APPROVED** [review]

File: tests/backup/backup_resiliency_test.go [hunk 3]
```diff
@@ -708,22 +708,43 @@ var _ = Describe("{CancelAllRunningBackupJobs}", func() {
...
 			log.Infof("The list of backups taken are: %v", backupNames)
-			log.InfoD("Sleeping for 20 seconds so that the request reaches stork and the backup creation process is started")
-			time.Sleep(20 * time.Second)
 		})
 		Step("Cancelling the ongoing backups", func() {
 			log.InfoD("Cancelling the ongoing backups")
 			ctx, err := backup.GetAdminCtxFromSecret()
 			log.FailOnError(err, "Fetching px-central-admin ctx")
+			requiredStatus := api.BackupInfo_StatusInfo_InProgress
+			expectedStatus := api.BackupInfo_StatusInfo_Pending
```
Comment (mkoppal-px) [inline tests/backup/backup_resiliency_test.go:717]: Name this variable as `backupInProgressStatus` and `backupPendingStatus`

**Pull request merged**